data/scripts/compress_images.sh

```diff
@@ -1,18 +1,40 @@
-for file in "../../public/images/members"/*.{jpg,png}; do
-    if [ -f "$file" ]; then
-        filename=$(basename -- "$file")
-        extension="${filename##*.}"
-        filename_no_ext="${filename%.*}"
-        out="../../public/images/members/$filename_no_ext.webp"
-
-        convert "$file" -resize 400x400 "$out"
-        if [ "$extension" != "webp" ]; then
-            rm "$file"
-        fi
-        echo "done"
-        echo "$filename"
+#!/bin/bash
+cd "$(git rev-parse --show-toplevel)"
+
+image_dir="public/images/members"
+members_file="data/members.ts"
+
+for file in "$image_dir"/*.{jpg,png,webp}; do
+  [ -e "$file" ] || continue  
+  if [ -f "$file" ]; then
+    filename=$(basename -- "$file")
+    extension="${filename##*.}"
+    filename_no_ext="${filename%.*}"
+    output_file="$image_dir/$filename_no_ext.webp"
+
+    convert "$file" -resize 400x400 "$output_file"
+    if [ "$extension" != "webp" ]; then
+      rm "$file"
     fi
+    echo "done"
+    echo "$filename"
+  fi
 done
 
-sed -i 's/\(jpg\|png\)/webp/g' ../members.ts
+echo "Updating photo extensions in $members_file..."
+
+# removed sed, updates photo extensions in members.ts using Node.js 
```

Comment:
nitpick 2: was there a reason for replacing `sed` here?
(edit: from discord I suspect you might be running into https://stackoverflow.com/a/4247319 - maybe try doing https://stackoverflow.com/a/12156783)